pkg/ko/publish/daemon.go

```diff
@@ -47,14 +48,23 @@ func (d *demon) Publish(img v1.Image, s string) (name.Reference, error) {
 	if err != nil {
 		return nil, err
 	}
+
+	for _, tagName := range d.tags {
+		tag, err := name.NewTag(fmt.Sprintf("%s/%s:%s", LocalDomain, d.namer(s), tagName), name.WeakValidation)
+		if err != nil {
+			return nil, err
+		}
+		log.Printf("Loading %v", tag)
+		if _, err := daemon.Write(tag, img); err != nil {
```

Comment:
This is going to effectively "docker load" the tarball multiple times, which is slow. For this, we probably want to expose a new method in the `daemon` package to add a tag that just calls [`ImageTag`](https://github.com/moby/moby/blob/beef00cb2612754d73024fd89d417b3a7114f87f/client/image_tag.go#L12). I think that's out of scope for this PR, but can you just add a TODO? Something like:

`TODO: This is slow because we have to load the image multiple times. We should use client.ImageTag instead`

cc @nkubala does that sound right to you?

We could also possibly hide this by caching the result of `ImageLoad` via `map[v1.Image]<something>` and calling `ImageTag` instead of `ImageLoad` if the `v1.Image` matches... :man_shrugging:

Reply:
This is something that i thought while doing it, but I had the conclusion that it should be an improvement for another time. 
The same is valid to the `default.go` when doing `remote.Write`

Reply:
@jonjohnsonjr  on the `default.go` we should be able to do this with `crane`? Just want to know to add a more specific TODO.

Reply:
I guess there is not a way to tag an image, but we can generalize the commit function to do multiple pushes of the manifest for each tag.

Reply:
I think we can optimize the `remote.Write` path a little bit to make it cheaper to add a tag to an existing image. We could also just expose a cheap "AddTag" function to do what is expected. Either way, I'm fine with just this TODO for now, we can revisit `remote.Write` later. It's much cheaper than the daemon side-loading thing (I think).